src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -883,32 +855,25 @@ else if (isWeightSumZero) {
      * <p>
      * This includes the penalties from static code analysis and of submission policies.
      *
-     * @param programmingExercise the participation belongs to.
-     * @param participation for which should be checked for possible penalties.
-     * @param staticCodeAnalysisFeedback automatic feedback from static code analysis.
-     * @param applySubmissionPolicy determines if the submission policy should be applied.
      * @return a total penalty that should be deducted from the score.
      */
-    private double calculateTotalPenalty(final ProgrammingExercise programmingExercise, final Participation participation, final List<Feedback> staticCodeAnalysisFeedback,
-            boolean applySubmissionPolicy) {
+    private double calculateTotalPenalty(ScoreCalculationData scoreCalculationData, boolean applySubmissionPolicy) {
         double penalty = 0;
-
-        int maxStaticCodeAnalysisPenalty = Optional.ofNullable(programmingExercise.getMaxStaticCodeAnalysisPenalty()).orElse(100);
-        if (Boolean.TRUE.equals(programmingExercise.isStaticCodeAnalysisEnabled()) && maxStaticCodeAnalysisPenalty > 0) {
-            penalty += calculateStaticCodeAnalysisPenalty(staticCodeAnalysisFeedback, programmingExercise);
+        var exercise = scoreCalculationData.exercise();
```

Comment:
When we use var, we can't use code completion. I guess its okay for scoreCalculationData but keep it in mind.

Reply:
The type is inferred from the type annotation of the function.
You still have autocompletion when using the variable further down